nullptr undeclared identifier when compiling using docker on github codespaces

### Prerequisites

- [x] I have searched for similar issues and confirmed this is not a duplicate

### Game Version

- [ ] Command & Conquer Generals
- [ ] Command & Conquer Generals: Zero Hour
- [x] Other (please specify below)

### Bug Description

The Docker build fails when running in a fresh GitHub Codespace. The error C2065: 'nullptr' : undeclared identifier appears multiple times during compilation.

It seems the recent code updates #1938  introduced nullptr (C++11), which is not supported by the Visual Studio 6 compiler (cl.exe) used in the Docker container.

Expected behavior: The ./scripts/docker-build.sh script should complete successfully as described in the "Quick Start" documentation.

### Reproduction Steps

Open https://github.com/TheSuperHackers/GeneralsGameCode
Press on <> Code
Press on Codespaces
Press on +
Wait till the vscode appears
Open terminal

type in: ./scripts/docker-build.sh

error log:
[@githubawn ➜ workspacesGeneralsGame.txt](https://github.com/user-attachments/files/24798152/%40githubawn.workspacesGeneralsGame.txt)


### Additional Context

_No response_

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,6 +72,13 @@ if (IS_VS6_BUILD)
     add_subdirectory(Dependencies/MaxSDK)
 endif()
 add_subdirectory(Dependencies/Utility)
+
+# To be removed when abandoning VC6
+if (IS_VS6_BUILD)
+    include_directories(Dependencies/Utility)
+    add_compile_options(/FIUtility/CppMacros.h)
+endif()
+
 add_subdirectory(resources)
 
 add_subdirectory(Core)

